src/qs_moments.F

```diff
@@ -212,98 +205,101 @@ SUBROUTINE dipole_velocity_deriv(qs_env, difdip, order, lambda, rc)
          IF (.NOT. ASSOCIATED(basis_set_b)) CYCLE
 
          ! basis ikind
-         first_sgfa => basis_set_a%first_sgf
-         la_max => basis_set_a%lmax
-         la_min => basis_set_a%lmin
-         npgfa => basis_set_a%npgf
-         nseta = basis_set_a%nset
-         nsgfa => basis_set_a%nsgf_set
-         rpgfa => basis_set_a%pgf_radius
-         set_radius_a => basis_set_a%set_radius
-         sphi_a => basis_set_a%sphi
-         zeta => basis_set_a%zet
-         ! basis jkind
-         first_sgfb => basis_set_b%first_sgf
-         lb_max => basis_set_b%lmax
-         lb_min => basis_set_b%lmin
-         npgfb => basis_set_b%npgf
-         nsetb = basis_set_b%nset
-         nsgfb => basis_set_b%nsgf_set
-         rpgfb => basis_set_b%pgf_radius
-         set_radius_b => basis_set_b%set_radius
-         sphi_b => basis_set_b%sphi
-         zetb => basis_set_b%zet
-
-         IF (inode == 1) last_jatom = 0
-
-         ! this guarantees minimum image convention
-         ! anything else would not make sense
-         IF (jatom == last_jatom) THEN
-            CYCLE
-         END IF
-
-         last_jatom = jatom
-
-         irow = iatom
-         icol = jatom
-
-         DO i = 1, 3
-         DO j = 1, 3
-            NULLIFY (mint(i, j)%block)
-            CALL dbcsr_get_block_p(matrix=difdip(i, j)%matrix, &
-                                   row=irow, col=icol, BLOCK=mint(i, j)%block, &
-                                   found=found)
-            CPASSERT(found)
-            mint(i, j)%block = 0._dp
-         END DO
-         END DO
+         ASSOCIATE ( &
```

Comment:
Should the ASSOCIATE construct also be used here?

Reply:
I think it does make sense here. I am aware that your former suggestion corresponds to the legacy coding style in CP2K. But with the ASSOCIATE construct it is clear that the variables are runtime constants.

Reply:
Okay, I replaced the pointer assignments in all files that are touched by this PR.